### PR TITLE
COL-29 - Follow up on outstanding code review

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
https://jira.ets.berkeley.edu/jira/browse/COL-29

### CODE REVIEW

**Comment by nicolaasmatthijs** [comment]
This PR builds on top of #10

**Comment by nicolaasmatthijs** [comment]
Rebased on master

**Comment by nicolaasmatthijs** [comment]
@paulkerschen : This might be another good PR for you to have a look at.

File: node_modules/col-course/lib/api.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ var getOrCreateCourse = module.exports.getOrCreateCourse = function(canvasCourse
   DB.Course.findOrCreate(options).complete(function(err, data) {
     if (err) {
       log.error({'err': err}, 'Failed to get or create a course');
-      return callback(err);
+      return callback({'code': 500, 'msg': err.message});
```
Comment (paulkerschen) [inline node_modules/col-course/lib/api.js]: @nicolaasmatthijs : just out of curiosity, is status code + message a convention for error objects in Node, or do people build them differently in different contexts?

File: node_modules/col-lti/tests/test-lti.js [hunk 2]
```diff
@@ -79,22 +79,32 @@ describe('LTI', function() {
...
       it('needs the correct lti credentials', function(callback) {
         var client = TestsUtil.getAnonymousClient();
         var course = TestsUtil.generateCourse(global.tests.canvas.ucberkeley);
         var user = TestsUtil.generateUser(global.tests.canvas.ucberkeley);
 
-        // The wrong LTI key will cause the Canvas instance to not be found
-        course.canvas.lti_key = '12345678901234567890123456789012';
-        LtiTestsUtil.assertAssetLibraryLaunchFails(client, course, user, 404, function() {
+        // An empty LTI key will cause validation to fail
+        course.canvas.lti_key = '';
```
Comment (paulkerschen) [inline node_modules/col-lti/tests/test-lti.js]: @nicolaasmatthijs Is this also expected behavior on null or undefined keys?

File: node_modules/col-course/lib/api.js (hunk 1 above)
Comment (nicolaasmatthijs) [inline node_modules/col-course/lib/api.js]: It's definitely not a widespread convention, although some projects do use it. Its main convenience is that the errors will be easy to deal with once they bubble up to the REST layer and the advantage of speaking HTTP error codes in the error objects is that they usually give quite a bit of information about the nature of the error.

File: node_modules/col-lti/tests/test-lti.js (hunk 2 above)
Comment (nicolaasmatthijs) [inline node_modules/col-lti/tests/test-lti.js]: I've added tests for null LTI keys and secrets.

**Comment by nicolaasmatthijs** [comment]
Followed up on review. Back to @paulkerschen for review and merge

**Pull request merged**